presto-jdbc/src/test/java/io/prestosql/jdbc/TestJdbcConnection.java

```diff
@@ -269,4 +293,35 @@ private static void assertConnectionSource(Connection connection, String expecte
             }
         }
     }
+
+    private static class ExtraCredentialsSystemTable
+            implements SystemTable
+    {
+        private static final SchemaTableName NAME = new SchemaTableName("test", "extraCredentials");
+
+        public static final ConnectorTableMetadata METADATA = tableMetadataBuilder(NAME)
+                .column("name", createUnboundedVarcharType())
+                .column("value", createUnboundedVarcharType())
+                .build();
+
+        @Override
+        public Distribution getDistribution()
+        {
+            return ALL_NODES;
+        }
+
+        @Override
+        public ConnectorTableMetadata getTableMetadata()
+        {
+            return METADATA;
+        }
+
+        @Override
+        public RecordCursor cursor(ConnectorTransactionHandle transactionHandle, ConnectorSession session, TupleDomain<Integer> constraint)
+        {
+            InMemoryRecordSet.Builder table = InMemoryRecordSet.builder(METADATA);
+            session.getIdentity().getExtraCredentials().entrySet().stream().forEach(credential -> table.addRow(credential.getKey(), credential.getValue()));
```

Comment:
`Map` has a nice `forEach` method which is great because you can name the key/value directly:
```java
session.getIdentity().getExtraCredentials().forEach((key, value) ->
        table.addRow(key, value));
```
Although in this case, we can use a method reference:
```java
session.getIdentity().getExtraCredentials().forEach(table::addRow);
```